mm/2s2h/Rando/ActorBehavior/ObjGrass.cpp

```diff
@@ -999,30 +990,26 @@ float roundUpToTwoDecimalPlaces(float value) {
 
 RandoCheckId IdentifyGrass(Vec3f pos) {
     RandoCheckId randoCheckId = RC_UNKNOWN;
-    pos.x = roundUpToTwoDecimalPlaces(pos.x);
-    pos.z = roundUpToTwoDecimalPlaces(pos.z);
+    pos.x = roundToWholeNumber(pos.x);
+    pos.z = roundToWholeNumber(pos.z);
 
-    if (gPlayState->sceneId != SCENE_KAKUSIANA) {
-        auto it = overworldGrassMap.find((SceneId)gPlayState->sceneId);
-        if (it == overworldGrassMap.end()) {
-            return randoCheckId;
-        }
-        auto innerIt = it->second.find({ pos.x, pos.z });
-        if (innerIt == it->second.end()) {
-            return randoCheckId;
-        }
-        return innerIt->second;
-    } else {
-        auto it = grottoGrassMap.find(gSaveContext.respawn[RESPAWN_MODE_UNK_3].data);
-        if (it == grottoGrassMap.end()) {
-            return randoCheckId;
-        }
-        auto innerIt = it->second.find({ pos.x, pos.z });
-        if (innerIt == it->second.end()) {
-            return randoCheckId;
-        }
-        return innerIt->second;
+    const auto& grassMap = (gPlayState->sceneId != SCENE_KAKUSIANA) ? overworldGrassMap : grottoGrassMap;
+    uint32_t key = (gPlayState->sceneId != SCENE_KAKUSIANA) ? static_cast<uint16_t>(gPlayState->sceneId)
+                   : (gSaveContext.save.entrance == 5184 || gSaveContext.save.entrance == 5280)
```

Comment:
Better to use entrance value builder

```suggestion
                   : (gSaveContext.save.entrance == ENTRANCE(GROTTOS, 4) || gSaveContext.save.entrance == ENTRANCE(GROTTOS, 10))
```